starter-spring/src/main/java/io/micrometer/prometheus/rsocket/autoconfigure/PrometheusRSocketClientAutoConfiguration.java

```diff
@@ -36,11 +36,11 @@
 public class PrometheusRSocketClientAutoConfiguration {
 
   @ConditionalOnMissingBean
-  @Bean(destroyMethod = "pushAndClose")
+  @Bean(destroyMethod = "pushAndCloseBlockingly")
   PrometheusRSocketClient prometheusRSocketClient(PrometheusMeterRegistry meterRegistry, PrometheusRSocketClientProperties properties) {
     return PrometheusRSocketClient.build(meterRegistry, properties.createClientTransport())
         .retry(Retry.backoff(properties.getMaxRetries(), properties.getFirstBackoff())
             .maxBackoff(properties.getMaxBackoff()))
-        .connect();
+        .connectBlockingly();
```

Comment:
Since this could cause previously quickly exiting applications to block for some time, maybe we should add some log at info or debug about connecting and the final push. I'm not sure whether we want to go so far as adding a log message saying "connected to prometheus-rsocket-proxy server in X ms" or just a message to let people know what is happening "Connecting to prometheus-rsocket-proxy server. This could take some time." and similar messages for the final push.

In theory, hopefully people would have recognized that their metrics were being lost before. But if they weren't, they may be surprised when they update versions and their application that was finishing in 100ms now takes 400ms or whatever if there are no log messages explaining what's taking time if it isn't their own code.